queries.sql

```diff
@@ -20,4 +20,57 @@ SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+/*
+  *Transactions actions
+*/
+-- Set the species column and then rollback
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT species FROM animals;
+ROLLBACK;
+SELECT species FROM animals;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE name NOT LIKE '%mon';
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+-- Delete all records from the animal TABLE
+BEGIN
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+
+-- Create SavePoint transaction
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT animalDeleteSavePoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO animalDeleteSavePoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 
+COMMIT;
+
+
+-- Query Aggrecation
+SELECT COUNT(*) FROM animals;
+
+--What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+
+
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=How%20many%20animals%20have%20never%20tried%20to%20escape%3F), Please add a query that answers the question `How many animals have never tried to escape?`. 
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Who%20escapes%20the%20most%2C%20neutered%20or%20not%20neutered%20animals%3F), Please add a query that answers the question `Who escapes the most, neutered or not neutered animals?`.

Reply:
# Added both queries
![image](https://user-images.githubusercontent.com/46789910/196498771-7f8ca388-cbd4-45a9-bb0d-bcd859e0dc64.png)